### ISSUE TITLE
Ranged creature AI won't attack target behind riot shield

### ISSUE THREAD

**Issue by itchyOwl** [FakeFishGames/Barotrauma#14877]
### Discussed in https://github.com/FakeFishGames/Barotrauma/discussions/14871

<div type='discussions-op-text'>

<sup>Originally posted by **A2ur3** October 20, 2024</sup>
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

https://github.com/user-attachments/assets/4f57848a-a53c-49de-a04e-30d2925d88be

### Reproduction steps

1. spawn creature Piratehuman01 from mod EuropaWar
2. hide behind a riot shield
3. creature wont attack even they can destroy the riot shield after a few shots

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

_No response_</div>

**Comment by Regalis11** [comment]
Tested the provided mod. 

It seems Piratehuman01 will now fire at the riot shield. I also tested making the riot shield indestructible, and noticed the enemy will keep firing at it regardless. Taking a look at this with the debugger, it seems the logic never gets to the new item check, because the raycast is only done to the collision categories Character, Wall and Level, and will never hit any items.

What's confusing here though is how/why the character didn't shoot at the riot shield. I gave the issue a test in the winter-update branch, and the enemy seems to fire at the shield even without the fix in this branch.

**Comment by itchyOwl** [comment]
> and the enemy seems to fire at the shield even without the fix in this branch

Hmm.. Human AI did that, but enemy AI not (if you drop the shield).

**Comment by itchyOwl** [comment]
> What's confusing here though is how/why the character didn't shoot at the riot shield before the fix. I gave the issue a test in the winter-update branch, and the enemy seems to fire at the shield even without the fix in this branch.

I can't repro this: If I drop the shield on the groun, the enemy stops shooting. The debugger says the ray hit a riot shield and the `Aim() `method retuns `false`.
![image](https://github.com/user-attachments/assets/e3fe1244-eab1-4e9e-95ff-93d45a3aa15f)

After the fix, the pirate shoots at the shield and also the debugger confirms it:
![image](https://github.com/user-attachments/assets/9319d327-58d7-483d-bb2e-42f2cf79b629)

Then I modified the riotshield to be non-damageable by projectiles. And it seems that now the raycast indeed doesn't hit the item. Seems to be because items that can be damaged are set in the collision category `CollisionCharacter` in Item:1180:
![image](https://github.com/user-attachments/assets/68afa4df-72e7-4664-b7aa-9532f21e98ba)

If you read the note there (or test it), you'll realize that items that can't be damaged by anything won't block anything. So enemies will just shoot through them, as they should. (And yeah I didn't remember this either)

Modifying the riotshield to be damaged by melee weapons will make it block the projectiles again, and also prevents the enemy to shoot at it. as expected.

In short, I don't think there's anything to fix here. Or actually there is a minor thing: riot shields don't seem to be damageable by meleeweapons or repair tools, which is probably an overisight. Addressed that in e5c3c7f.

**Comment by Regalis11** [comment]
> If you read the note there (or test it), you'll realize that items that can't be damaged by anything won't block anything. So enemies will just shoot through them, as they should. (And yeah I didn't remember this either)

I don't think this is true. The thing that the projectiles hit isn't the item's actual collider (that's disabled when the item is held), it's the extra "Pusher" collider created by `Holdable`.

I'm also not talking about dropped items, I'm talking about the the shield being held. It seems the bot fires at the shield regardless if it's damageable by projectiles or not, because the raycast cannot hit the item (since the collider is in the `CollisionItemBlocking` category, which we're not checking in the raycast).

Here's a clip of how the enemy reacts to an indestructible shield: they keep firing at it, even though they shouldn't?

https://github.com/user-attachments/assets/10748e4f-ecf4-455d-8e6e-a5d8ad5c7e6c

**Comment by itchyOwl** [comment]
> I'm also not talking about dropped items

The issue was specifically about dropped items, so you should have said the issue you spotted was specifically about _held_ items.
